middleware/middleware.go

```diff
@@ -81,20 +82,21 @@ func (l *logHandler) applyHandlers(req *http.Request, finalizer map[string]inter
 	return result
 }
 
-// New takes in an http Handler to wrap with logging, the logger to use, and any amount of
+// New takes in an http Handler to wrap with logging, the logger source name to use, and any amount of
 // optional handlers to customize the data that's logged.
-func New(h http.Handler, logger *logger.Logger, handlers ...func(*http.Request) map[string]interface{}) http.Handler {
+// On every request, the middleware will create a logger and place it in req.Context().
+func New(h http.Handler, source string, handlers ...func(*http.Request) map[string]interface{}) http.Handler {
```

Comment:
Catapult uses v4 for other reasons, and doesn't make use of the logger context at all (so is unaffected).  I'd like for this change not to make builds there start failing.